stdlib/public/core/ObjectIdentifier.swift

```diff
@@ -14,6 +14,14 @@
 ///
 /// In Swift, only class instances and metatypes have unique identities. There
 /// is no notion of identity for structs, enums, functions, or tuples.
+///
+/// `ObjectIdentifier` is only guaranteed to remain unique for the
+/// lifetime of an object. When the instance gets deallocated, its object 
+/// identifier may be reused for a different object. (Internally, objects are
+/// identified by their memory location.)
+///
+/// If you need an object identifier over the lifetime of an object, it may
+/// be appropriate to provide a custom implementation of `Identifiable`.
```

Comment:
I don't think this final paragraph is correct.

Firstly, it may be misleading, because a custom implementation of `Identifiable` won't _directly_ affect `ObjectIdentifier`.

Secondly, it should say something more like "a _unique_ object identifier over the lifetime of a _program_".